code/modules/library/soapstone.dm

```diff
@@ -154,12 +154,42 @@ but only permanently removed with the curator's soapstone.
 
 /obj/structure/chisel_message/update_icon()
 	. = ..()
-	var/hash = md5(hidden_message)
-	var/newcolor = copytext_char(hash, 1, 7)
-	add_atom_colour("#[newcolor]", FIXED_COLOUR_PRIORITY)
-	set_light_color("#[newcolor]")
+
+	var/newcolor = COLOR_SOAPSTONE_IRON
+	switch(like_keys.len - dislike_keys.len)
+		if(-INFINITY to -1)
+			newcolor = COLOR_SOAPSTONE_PLASTIC
+		if(0 to 14)
+			newcolor = COLOR_SOAPSTONE_IRON
+		if(15 to 29)
+			newcolor = COLOR_SOAPSTONE_BRONZE
+		if(30 to 44)
+			newcolor = COLOR_SOAPSTONE_SILVER
+		if(45 to 59)
+			newcolor = COLOR_SOAPSTONE_GOLD
+		if(60 to INFINITY)
+			newcolor = COLOR_SOAPSTONE_DIAMOND
```

Comment:
It's probably over engineering to deduplicate this logic in update_icon and update_name (especially because I explicitly asked for them to be split up).

Wish DM had something like returning 2-tuples, so you could have the determination logic separate from what actually sets the colour and name.

You could make a map (colour -> name), have a proc to determine the colour, and then have `update_name()` use that map...

Reply:
You'd have to give me an example of what a map looks like. Sorry, I don't know a lot about coding.

Reply:
@coiax So what's going on?

Reply:
@cacogen Could we at least get defines for the range values please

Reply:
can he just explain what he means by a map